stackslib/src/net/download/nakamoto/tenure_downloader.rs

```diff
@@ -651,10 +643,9 @@ impl NakamotoTenureDownloader {
         &mut self,
         response: StacksHttpResponse,
     ) -> Result<Option<Vec<NakamotoBlock>>, NetError> {
-        self.idle = true;
```

Comment:
> Is the change here intended to make it so that self.idle is set to true after this function has successfully run, but not if it errors?

Yes; this had not been the case before this patch.

> Does it matter that the self.idle remains in the "old state" for error conditions?

That's the intent -- the downloader set will re-try the next download step if it can.

I can make that refactoring